lib/route/route_obj.c

```diff
@@ -284,6 +289,9 @@ static void route_dump_details(struct nl_object *a, struct nl_dump_params *p)
 			r->rt_ttl_propagate ? "enabled" : "disabled");
 	}
 
+	if (r->ce_mask & ROUTE_ATTR_NHID && r->rt_nhid != 0)
```

Comment:
there needs to be consistency about what `rt_nhid=0` means.

Note that to kernel (`ip nexthop add id 0 dev eth0`), zero ID means to choose an unused ID.

Below, `rtnl_route_set_nhid()` accepts setting to 0, while also setting the `ROUTE_ATTR_NHID`.


What needs to be done, is that the code never checks for `r->rt_nhid != 0` directly. Instead, this is a ID-based next-hop if-and-only-if `ROUTE_ATTR_NHID` is set.

In this case, during dump we want to see when the nhid is set to zero and not hide that (important) fact.